numba/utils.py

```diff
@@ -77,11 +78,18 @@ def erase_traceback(exc_value):
     from inspect import Parameter as pyParameter
 except ImportError:
     try:
-        from funcsigs import signature as pysignature
+        from funcsigs import signature as _pysignature
         from funcsigs import Signature as pySignature
         from funcsigs import Parameter as pyParameter
         from funcsigs import BoundArguments
 
+        def pysignature(*args, **kwargs):
+            try:
+                return _pysignature(*args, **kwargs)
+            except ValueError:
+                msg = ("Cannot obtain a signature for:", args)
+                raise UnsupportedError(msg)
```

Comment:
Better to include the original message, which will contain " '.0' is not a valid parameter name". 

Also,  it is weird that python2 is generating that a arg with `.0` as the name, and that `funcsigs` can't handle it.

Reply:
> Better to include the original message, which will contain " '.0' is not a valid parameter name".

Good idea

> Also, it is weird that python2 is generating that a arg with `.0` as the name, and that `funcsigs` can't handle it.

Yes, this was weird. That you have seen the `.0` confirms it is not just me. Looking at the byte code disassembly, it's definitely in there!